### ISSUE TITLE
eqmx_auth_http plugin doesn't work in docker

### ISSUE THREAD

**Issue by gbaranski** [emqx/emqx#3731]
hello, thats how docker-compose file looks like

```yml
  emqx:
    image: 'emqx/emqx'
    container_name: 'emqx'
    volumes:
      - ./emqx/data:/opt/emqx/data
    environment:
      - EMQX_NAME=emqx
      - EMQX_LOADED_PLUGINS=emqx_recon,emqx_retainer,emqx_management,emqx_dashboard,emqx_auth_http

      - EMQX_ALLOW_ANONYMOUS=false

      # # ------------------------------
      - EMQX_AUTH__HTTP__AUTH_REQ=http://server-api:8000/auth/user
      - EMQX_AUTH__HTTP__AUTH_REQ__METHOD=POST
      - EMQX_AUTH__HTTP__AUTH_REQ__PARAMS=clientId=%c,username=%u,password=%P,ipaddr=%a,protocol=%r
      - EMQX_AUTH__HTTP__AUTH_REQ__CONTENT_TYPE=JSON

      # ------------------------------
      - EMQX_AUTH__HTTP__ACL_REQ=http://server-api:8000/auth/acl
      - EMQX_AUTH__HTTP__ACL_REQ__METHOD=POST
      - EMQX_AUTH__HTTP__ACL_REQ__CONTENT_TYPE=JSON
      - EMQX_AUTH__HTTP__ACL_REQ__PARAMS=access=%A,username=%u,clientId=%c,ipaddr=%A,topic=%t,mountPoint=%m
    ports:
      - 1883:1883 # MQTT
      - 8083:8083
      - 8883:8883
      - 8084:8084
      - 18083:18083

    networks:
      - backend
```

when i start container everything looks fine, but when I go to dashboard, http_auth plugin looks like disabled, also when I try to connect to broker it just says `login failed for not_authorized`  and no request is sent to `server-api` container.

When i try to enable plugin in dashboard it just says
```
emqx          | (emqx@172.22.0.3)1> 2020-09-20 09:50:40.287 [alert] [Plugins] Plugin emqx_auth_http load failed with {function_clause,
emqx          |                                                   [{emqx_plugins,
emqx          |                                                     apply_configs,
emqx          |                                                     [{error,
emqx          |                                                       transform_datatypes,
emqx          |                                                       {errorlist,
emqx          |                                                        [{error,
emqx          |                                                          {transform_type,
emqx          |                                                           "auth.http.acl_req.content_type"}},
emqx          |                                                         {error,
emqx          |                                                          {enum_name,
emqx          |                                                           {"JSON",
emqx          |                                                            ["json",
emqx          |                                                             "x-www-form-urlencoded"]}}},
emqx          |                                                         {error,
emqx          |                                                          {transform_type,
emqx          |                                                           "auth.http.auth_req.content_type"}},
emqx          |                                                         {error,
emqx          |                                                          {enum_name,
emqx          |                                                           {"JSON",
emqx          |                                                            ["json",
emqx          |                                                             "x-www-form-urlencoded"]}}},
emqx          |                                                         {error,
emqx          |                                                          {transform_type,
emqx          |                                                           "auth.http.auth_req.method"}},
emqx          |                                                         {error,
emqx          |                                                          {enum_name,
emqx          |                                                           {"POST",
emqx          |                                                            ["post",
emqx          |                                                             "get"]}}}]}}],
emqx          |                                                     [{file,
emqx          |                                                       "/emqx-rel/_checkouts/emqx/src/emqx_plugins.erl"},
emqx          |                                                      {line,240}]},
emqx          |                                                    {emqx_plugins,load_plugin,
emqx          |                                                     2,
emqx          |                                                     [{file,
emqx          |                                                       "/emqx-rel/_checkouts/emqx/src/emqx_plugins.erl"},
emqx          |                                                      {line,263}]},
emqx          |                                                    {emqx_mgmt_api_plugins,
emqx          |                                                     load,2,
emqx          |                                                     [{file,
emqx          |                                                       "/emqx-rel/_checkouts/emqx_management/src/emqx_mgmt_api_plugins.erl"},
emqx          |                                                      {line,80}]},
emqx          |                                                    {minirest_handler,
emqx          |                                                     dispatch,2,
emqx          |                                                     [{file,
emqx          |                                                       "/emqx-rel/_checkouts/minirest/src/minirest_handler.erl"},
emqx          |                                                      {line,84}]},
emqx          |                                                    {minirest,handle_request,
emqx          |                                                     2,
emqx          |                                                     [{file,
emqx          |                                                       "/emqx-rel/_checkouts/minirest/src/minirest.erl"},
emqx          |                                                      {line,112}]},
emqx          |                                                    {minirest,init,2,
emqx          |                                                     [{file,
emqx          |                                                       "/emqx-rel/_checkouts/minirest/src/minirest.erl"},
emqx          |                                                      {line,104}]},
emqx          |                                                    {cowboy_handler,execute,2,
emqx          |                                                     [{file,
emqx          |                                                       "/emqx-rel/_checkouts/cowboy/src/cowboy_handler.erl"},
emqx          |                                                      {line,41}]},
emqx          |                                                    {cowboy_stream_h,execute,
emqx          |                                                     3,
emqx          |                                                     [{file,
emqx          |                                                       "/emqx-rel/_checkouts/cowboy/src/cowboy_stream_h.erl"},
emqx          |                                                      {line,320}]}]}
```

**Comment by HJianBo** [comment]
Hi @gbaranski Please replace this line
```
- EMQX_AUTH__HTTP__AUTH_REQ__CONTENT_TYPE=JSON
```

with
```
- EMQX_AUTH__HTTP__AUTH_REQ__CONTENT_TYPE=json
```

**Comment by gbaranski** [comment]
@HJianBo same issue, no request is sent to EMQX_AUTH__HTTP__AUTH_REQ

**Comment by gbaranski** [comment]
is there any way i can verify if plugin is running?

**Comment by HJianBo** [comment]
Use the `bin/emqx_ctl plugins list|grep emqx_auth_http` command, e.g:
```
Plugin(emqx_auth_http, description=EMQ X Authentication/ACL with HTTP API, active=false)
```

The `active` indicates the plugin running state

**Comment by gbaranski** [comment]
@HJianBo okay, for some reason it doesn't start by itself, and I can't load him because of 
`Load plugin emqx_auth_http error: parse_config_file_failed.`

**Comment by gbaranski** [comment]
Issue resolved by copying whole /opt/emqx/etc/plugins to some directory, and making a volume to that, but I think that needs some fix, or at least use default configurations if they are not present in volume

**Comment by redwolf2019** [comment]
@gbaranski same issue, i switch to `emqx/emqx:v4.1.4-alpine-amd64`, eqmx_auth_http plugin was runing normally.

**Comment by zmstone** [comment]
the key is `web.hook.headers.content-type`
so I don't think environment variable overrides will work because of the dash in the config key.

**Comment by zmstone** [comment]
merged just about the right time. the fix will be in v4.3.0
For older versions: config keys with `-` in them do not support environment variable override.
